### PR TITLE
DEX-515 Change social-groups PATCH to only use /members

### PR DESCRIPTION
**Pull request by karenc**
- Add social group integration test

- Change social-groups PATCH to only use /members

  Remove `/member` PATCH path and instead use `/members` to be consistent.
  When op is "add", the members are added (or replaced if they already
  existed).  When the op is "removed", the member or list of members are
  removed.  When the op is "replaced", the members are set to exactly what
  the user provides.

- Fix error when social group member is not valid uuid

  ```
  E               sqlalchemy.exc.StatementError: (builtins.ValueError) badly formed hexadecimal UUID string
  E               [SQL: SELECT individual.created AS individual_created, individual.updated AS individual_updated, individual.viewed AS individual_viewed, individual.guid AS individual_guid, individual.featured_asset_guid AS individual_featured_asset_guid, individual.version AS individual_version
  E               FROM individual
  E               WHERE individual.guid = %(param_1)s]
  E               [parameters: [{'%(140645722029696 param)s': '4037'}]]
  ```

### CODE REVIEW

File: app/modules/social_groups/parameters.py [hunk 1]
```diff
@@ -35,11 +34,26 @@ class PatchSocialGroupDetailsParameters(PatchJSONParameters):
...
+            from . import resources
+
+            members = {
+                str(member.individual_guid): {'roles': member.roles}
+                for member in obj.members
+            }
+            for member_guid, roles in value.items():
+                members[member_guid] = roles
+
+            resources.validate_members(members)
```
Comment (hwindsor) [inline app/modules/social_groups/parameters.py:51]: resources validate_member and validate_members were made separate functions to allow the usage below.
As that  has been removed, they can now be simplified.

File: app/modules/social_groups/parameters.py [hunk 2]
```diff
@@ -88,8 +80,11 @@ def replace(cls, obj, field, value, state):
     def remove(cls, obj, field, value, state):
         # Any researcher can remove a member in the group
         ret_val = False
-        if field == 'member':
-            ret_val = obj.remove_member(value)
-            msg = f'Removing member {value}'
+        if field == 'members':
+            if not isinstance(value, list):
+                value = [value]
```
Comment (hwindsor) [inline app/modules/social_groups/parameters.py:85]: members = [value] 
would be clearer than reusing the value variable, then member rather han single character v

File: integration_tests/test_social_groups.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    data = {
+        'key': 'social_group_roles',
+        'data': {
+            'Matriarch': {'multipleInGroup': False},
+            'Patriarch': {'multipleInGroup': True},
+        },
+    }
+    response = session.post(codex_url('/api/v1/site-settings/'), json=data)
+    assert response.status_code == 200
+    assert response.json() == {
```
Comment (hwindsor) [inline integration_tests/test_social_groups.py:14]: any reason you don't just check response.json() == data

File: integration_tests/test_social_groups.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            'Patriarch': {'multipleInGroup': True},
+        },
+        'created': response.json()['created'],  # 2021-10-29T20:17:46.585288+00:00
+        'string': response.json()['string'],  # '' or None
+        'file_upload_guid': None,
+        'public': True,
+        'updated': response.json()['updated'],
+    }
+
+    # Create encounters
```
Comment (hwindsor) [inline integration_tests/test_social_groups.py:27]: I can see a lot of integration tests wanting to create a sighting and an individual. 
These should be utils for reuse

**Review by hwindsor — APPROVED** [review]
Couple of minor changes requested. 
Also as API has changed, please update the doc:
https://docs.google.com/document/d/1d1L8j6vOy1h3tpcdl7qWQcn87gYkyDTsCzNISnEGVgM/edit#heading=h.i3s24ptpxrf4

**Review by naknomum — APPROVED** [review]
lgtm

File: integration_tests/test_social_groups.py (hunk 4 above)
Comment (karenc) [reply integration_tests/test_social_groups.py:27]: :+1: We can move them to utils when there's another test that uses the same code

File: app/modules/social_groups/parameters.py (hunk 1 above)
Comment (karenc) [reply app/modules/social_groups/parameters.py:51]: I merged `validate_member` into `validate_members` but that was about it :joy:

Comment (hwindsor) [reply app/modules/social_groups/parameters.py:51]: Ah yes, you did. Ignore me :)

Comment (karenc) [reply app/modules/social_groups/parameters.py:51]: No I just force pushed 🙂

File: app/modules/social_groups/parameters.py (hunk 2 above)
Comment (karenc) [reply app/modules/social_groups/parameters.py]: Fixed

**Comment by karenc** [comment]
> Also as API has changed, please update the doc: https://docs.google.com/document/d/1d1L8j6vOy1h3tpcdl7qWQcn87gYkyDTsCzNISnEGVgM/edit#heading=h.i3s24ptpxrf4

Done :+1:

**Pull request merged**